src/main/java/teammates/common/util/Url.java

```diff
@@ -1,15 +1,17 @@
 package teammates.common.util;
 
-public class Url {
+/**
+ * The Url class represents a URL string.
+ * It provides methods to manipulate the URL string and extract values from it.
+ */
+public abstract class Url {
```

Comment:
See my comment in Config class for an example of using setters. Yes, we may not need a setter for the baseUrl.
Yes, it makes sense for the relative path to have a leading slash. My point is, it should be specified in the API clearly.